struct.md

```diff
@@ -0,0 +1,310 @@
+[top VSS file](https://github.com/COVESA/vehicle_signal_specification/blob/master/spec/VehicleSignalSpecification.vspec).
+Tooling may in the future support overlays for type declaration similar to how it is supported for signals.
+
+**TBD: What naming restriction shall apply**
+
+Shall it be allowed to use the same branch names in the type tree as in the signal tree, or must they be totally separated?
+If we consider using standard VSS catalog where all signals resides in the `Vehicle` top branch,
+is it then allowed to call the top-level in the type tree `Vehicle` as well?
+That could be useful if we want to use paths like `Vehicle.Types.SomeType`for VSS standard catalog in the future.
+Or shall it be required to have a totally separate tree, e.g. starting with `Types`?
+
+Theoretically we could allow to have exactly the same branch structure in the signal files as in the type files and even reuse the same name (with full path)
+as it anyway is clear from context whether we refer to a type or not. I.e. theoretically we could allow the signal `Vehicle.A.B` to refer to the struct `Vehicle.A.B`
+
+*Alternative 1: Do not enforce any restrictions on syntactic/semantic level, i.e. tooling shall support any naming style of the types. This does not prevent us from agreeing that top level name shall be e.g. "Types" in the standard catalog*
+
+*Alternative 2: Require that top path for the type file must be "Types", i.e. tooling shall give error if another name is found!*
+
+
+## Simple Definition and Usage
+
+This could be a hypothetical content of a VSS type file
+
+```
+Types:
+  type: branch
+  
+Types.DeliveryInfo:
+  type: struct
+  description: A struct type containing info for each delivery
+  
+Types.DeliveryInfo.Address:
+  datatype: string
+  type: item
+  description: Destination address
+
+Types.DeliveryInfo.Receiver:
+  datatype: string
+  type: item
+  description: Name of receiver
+```
+
+This struct definition could then be referenced from the VSS signal tree
+
+```
+Delivery:
+  datatype: Types.DeliveryInfo
+  type: sensor
+```
+
+For VSS 4.0 it is not necessary that vss-tools do semantic check, i.e. if someone would add an extra `f` by mistake like this:
+
+```
+Delivery:
+  datatype: Types.DeliveryInffo
+  comment: Note: Spelling error on line above, will only be detected if semantic check is implemented
+  type: sensor
+```
+
+... then VSS-tools does not necessarily need to give an error (stretch goal to have semantic check that referred type exist).
+
+The type file may contain sub-branches and `#include`-statements just like regular VSS files
+
+```
+Types:
+  type: branch
+  
+Types.Powertrain:
+  type: branch
+  description: Powertrain types.
+#include Powertrain/Powertrain.vspec Types.Powertrain
+
+```
+
+## Name resolution
+
+For now, two ways of referring to a type shall be considered correct:
+
+* Reference by (leaf) name to a struct definition within a branch with the same name in the type tree
```

Comment:
This is no longer valid with the approach of hosting types in a separate tree right?

Reply:
There could be two scenarios where this could be relevant:

**Internal references within type tree**

This is valid and makes sense, that item B.B below can refer to type A within the same file without the need to the full path. I.e. supporting just using leaf name if referring to a type in the same branch in the same tree.

Example - Parser finds that B.B refers to type A. It will then check the branch containing struct B and find struct A and use it.

```
Struct A
  Item A.A
  Item A.B
Struct B
 Item B.A
 Item B.B: Datatype: A
```


**Same branch names in signal and type tree**

This is more of a corner case. If you use the same branch structure in signal tree and type tree, and you from signal tree refer to a type in a branch with the same name in the type tree - shall that be allowed?

Example - if signal A.B.C refers to type D, shall the parser then check if there exist a type A.B.D in the type tree?

Or shall we restrict leaf-name lookup to only be within the same tree, i.e. practically allowing it only for the type tree?

Signal Tree:

```
Branch A
  Branch B
    Signal C, Datatype = D;
```
Type tree:

```
Branch A
  Branch B
    Struct D;
```

Reply:
Hmm, this corner case makes me think that we need a dedicated root node name for types just like we have one for signals, namely, `Vehicle`

Reply:
I agree on the need for a dedicated root node name for types.